src/ui/public/vis/editors/config/editor_config_providers.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AggConfig } from '../..';
+import { AggType } from '../../../agg_types';
+import { IndexPattern } from '../../../index_patterns';
+import { leastCommonMultiple } from '../../../utils/math';
+import { EditorConfig, EditorParamConfig, FixedParam, NumericIntervalParam } from './types';
+
+type EditorConfigProvider = (
```

Comment:
could we use a different name for this ... Provider is currently used to indicate you will need to call it with Private to get the actual thing out of it.

Reply:
I struggled with that too. The problem is I couldn't come up with a better name (`Service` would sound more like the central class, and couldn't find much better names). We also just had a discussion with CJ and Jen, and couldn't find any better name. So I'll use CJ's argument here, that we are trying to remove `Private` in the long run and it shouldn't keep the monopoly on that word, since it makes sense in other places too where it's more conventional.